### PR TITLE
[DOCS] Reformat rollover index API docs

### PR DESCRIPTION
**Pull request by jrodewig**
Reformats the rollover API docs to align with the new [API reference template](https://github.com/elastic/docs/blob/master/shared/api-ref-ex.asciidoc).

Relates to #43765.

### CODE REVIEW

File: docs/reference/indices/rollover-index.asciidoc [hunk 1]
```diff
@@ -24,17 +56,102 @@ from the original (rolled-over) index.
...
-| max_docs | The maximum number of documents the index should contain. This does not add documents multiple times for replicas
-| max_size | The maximum estimated size of the primary shard of the index
-|===
+[[rollover-wait-active-shards]]
+===== Wait for active shards
+
+Because the rollover operation creates a new index to rollover to, the
+<<create-index-wait-for-active-shards,`wait_for_active_shards`>> setting on
+index creation applies to the rollover action as well.
+
```
Comment (debadair) [inline docs/reference/indices/rollover-index.asciidoc]: The "as well" isn't really necessary. (Total nit.)

File: docs/reference/indices/rollover-index.asciidoc [hunk 2]
```diff
@@ -24,17 +56,102 @@ from the original (rolled-over) index.
...
+==== {api-request-body-title}
+
+include::{docdir}/rest-api/common-parms.asciidoc[tag=aliases]
+
+`conditions`::
++
+--
+(Required, object)
+Set of conditions the index alias's existing index must met to rollover.
+
```
Comment (debadair) [inline docs/reference/indices/rollover-index.asciidoc]: I'd say "to roll over" rather than close it up. (Another nit.)

**Review by debadair — APPROVED** [review]
Left a couple minor editorial comments, but LGTM.

File: docs/reference/indices/rollover-index.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/indices/rollover-index.asciidoc]: Fixed with 58898c608379c37ef519bfe6433c4e674a939ce9

File: docs/reference/indices/rollover-index.asciidoc (hunk 2 above)
Comment (jrodewig) [reply docs/reference/indices/rollover-index.asciidoc]: Fixed with b39cfcc27b78c0bbf1a2a1f4cd5bc1a78459dc60

**Pull request merged**